recipes/gdal/pre_3.5.0/patches/3.4.x/fix-autotools-3.4.3.patch

```diff
@@ -0,0 +1,333 @@
+ 
+-CONFIG_LIBS	=	$(GDAL_ROOT)/$(LIBGDAL)
++CONFIG_LIBS	=	$(GDAL_ROOT)/$(LIBGDAL) $(LIBS)
+ ifeq ($(MACOSX_FRAMEWORK),yes)
+ CONFIG_LIBS_INS	=	-L$(INST_LIB)/unix/lib -lgdal
+ else
+--- a/configure.ac
++++ b/configure.ac
+@@ -1168,6 +1168,10 @@ AC_ARG_WITH(libdeflate,
+                             [enable libdeflate usage (optional for faster Deflate support (still requires zlib) (ARG=yes/no/prefix), enabled by default)]),,)
+ 
+ if test "$with_libdeflate" != "no" ; then
++# AC_CHECK_LIB tests not robust enough (don't handle static libdeflate on windows)
++if test "$with_libdeflate" = "yes" ; then
++  LIBDEFLATE_SETTING=yes
++else
+ 
+   if test "$with_libdeflate" != "" -a "$with_libdeflate" != "yes"; then
+     AC_CHECK_LIB(deflate, libdeflate_zlib_decompress, [libdeflate_lib=yes], [libdeflate_lib=no],-L$with_libdeflate/lib)
+@@ -1197,13 +1201,13 @@ if test "$with_libdeflate" != "no" ; then
+   fi
+ 
+ fi
++fi
+ 
+ if test "$LIBDEFLATE_SETTING" = "yes" ; then
+   if test "$with_libdeflate" != "" -a "$with_libdeflate" != "yes"; then
+     EXTRA_INCLUDES="-I$with_libdeflate/include $EXTRA_INCLUDES"
+     LIBS="-L$with_libdeflate/lib -ldeflate $LIBS"
+-  else
+-    LIBS="-ldeflate $LIBS"
++    # Too fragile: libdeflate static lib name on windows is libdeflatestatic
+   fi
+ fi
+ 
+@@ -1371,7 +1375,7 @@ else
+   AC_MSG_RESULT([using libtiff from ${with_libtiff}.])
+ 
+   dnl Check for the BigTIFF enabled library (libtiff >= 4.0)
+-  AC_CHECK_LIB(tiff,TIFFScanlineSize64,HAVE_BIGTIFF=yes,HAVE_BIGTIFF=no,)
++  AC_CHECK_LIB(tiff,TIFFScanlineSize64,HAVE_BIGTIFF=yes,HAVE_BIGTIFF=no,$LIBS)
+ 
+ fi
+ 
+@@ -1395,23 +1399,11 @@ AC_ARG_WITH(curl,
+ dnl Clear some cache variables
+ unset ac_cv_path_LIBCURL
+ 
+-if test "`basename xx/$with_curl`" = "curl-config" ; then
+-  LIBCURL_CONFIG="$with_curl"
+-elif test "$with_curl" = "no" ; then
+-  LIBCURL_CONFIG=no
++PKG_PROG_PKG_CONFIG([0.21])
++if test "$with_curl" = "yes" ; then
++  PKG_CHECK_MODULES([CURL], [libcurl], [CURL_SETTING=yes], [CURL_SETTING=no])
+ else
+-  AC_PATH_PROG(LIBCURL_CONFIG, curl-config, no)
+-fi
+-
+-if test "$LIBCURL_CONFIG" != "no" ; then
+-
+-  CURL_VERNUM=`$LIBCURL_CONFIG --vernum`
+-  CURL_VER=`$LIBCURL_CONFIG --version | awk '{print $2}'`
+-
+-  AC_MSG_RESULT([        found libcurl version $CURL_VER])
+-
+-  AC_CHECK_LIB(curl,curl_global_init,CURL_SETTING=yes,CURL_SETTING=no,`$LIBCURL_CONFIG --libs`)
+-
++  CURL_SETTING=no
+ fi
+ 
+ AC_SUBST(CURL_SETTING,$CURL_SETTING)
+@@ -1423,8 +1415,11 @@ dnl Check for SQLite
+ dnl Proj depends on it so it must appear before.
+ dnl ---------------------------------------------------------------------------
+ 
+-SQLITE3_REQ_VERSION="3.0.0"
+-AX_LIB_SQLITE3($SQLITE3_REQ_VERSION)
++HAVE_SQLITE3=no
++AC_ARG_WITH(sqlite3, AS_HELP_STRING([--with-sqlite3[=ARG]], [use SQLite 3 library (ARG=yes or no)]),,)
++if test x"$with_sqlite3" = x"yes" ; then
++    PKG_CHECK_MODULES([SQLITE3], [sqlite3 >= 3.0.0], [HAVE_SQLITE3=yes], [HAVE_SQLITE3=no])
++fi
```

Comment:
I think you can use the following pattern to be 100% sure that sqlite3 is actually found and used

```
+if test x"$with_sqlite3" = x"yes" ; then
+    PKG_CHECK_MODULES([SQLITE3], [sqlite3 >= 3.0.0], [HAVE_SQLITE3=yes], [AC_MSG_ERROR([sqlite3 not found]))
+else
+    HAVE_SQLITE3=no
+fi
```
But perhaps overkill since gdal on autotools x is soon to be deprecated

Reply:
ok, all patches updated for all dependencies found with `PKG_CHECK_MODULES` due to the patch.